Assets/Scripts/Behaviours/SteeringBehaviours.cs

```diff
@@ -18,42 +20,53 @@ public class SteeringBehaviours : MonoBehaviour
     [SerializeField] private Deceleration arriveDeceleration = Deceleration.medium;
     [SerializeField] private bool seekON = false;
     [SerializeField] private bool arriveOn = false;
-    [SerializeField] private bool persuitOn = false;
+    [SerializeField] private bool pursuitOn = false;
+    [SerializeField] private bool OffsetPursuitOn = false;
+    [SerializeField] private bool wallAvoidanceON = false;
+    [SerializeField] private BasicMovingEnemy Leader = null;
+    [SerializeField] private Vector2 OffsetToleader = new Vector2(2, 2);
+
+    private FeelerManager feelers;
+
 
     [SerializeField] private bool fleeON = false;
     [SerializeField] private bool evadeOn = false;
-    private double panicDistance = 2 * 2;
 
-    [SerializeField] private Collider2D headingCollider;
-    private bool collidingWall = false;
+    [SerializeField] private float arriveForce = 1f;
+    [SerializeField] private float fleeForce = 1f;
+    private double panicDistance = 2 * 2;
 
 
 
     private void Start()
     {
         host = GetComponent<BasicMovingEnemy>();
         wanderTarget = host.heading * wanderRadius;
-        wanderCircle = new Vector2();
-        worldTarget = new Vector2();
         rand = new System.Random();
+        feelers = GetComponentInChildren<FeelerManager>();
     }
 
     public Vector2 Calculate()
     {
         returnValue = Vector2.zero;
-
-        if (seekON) returnValue += Seek();
-        if (fleeON) returnValue += Flee();
-        if (arriveOn) returnValue += Arrive(arriveDeceleration);
-        if (persuitOn) returnValue += Persuit();
-        if (evadeOn) returnValue += Evade() * 2;
-        if (wanderOn) returnValue += Wander();
-        if (collidingWall) returnValue *= -1;
+        if (host.target != null)
+        {
+            if (wallAvoidanceON) returnValue += WallAvoidance();
+            if (seekON) returnValue += Seek();
+            if (fleeON) returnValue += Flee() * fleeForce;
+            if (arriveOn) returnValue += Arrive(arriveDeceleration) * arriveForce;
+            if (pursuitOn) returnValue += Pursuit();
+            if (evadeOn) returnValue += Evade() * 2;
+            if (wanderOn) returnValue += Wander();
+            if (OffsetPursuitOn) returnValue += OffsetPursuit();
+        }
+        else
+        {
+            returnValue += Wander();
+        }
```

Comment:
Maybe add this is an seperate class where you can easily change up the scores of seperate steeringbehaviours?